pkg/services/serviceaccounts/database/token_store.go

```diff
@@ -152,20 +154,32 @@ func (s *ServiceAccountsStoreImpl) UpdateAPIKeysExpiryDate(ctx context.Context,
 	if 1 > expiryDaysLimit {
 		return nil
 	}
-	err := s.sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
-		rawSQL := `UPDATE api_key
-							 SET expires = CAST(strftime('%s', created) AS INT) - (CAST(strftime('%s', created) AS INT) % 86400) + ((? + 1) * 86400),
-							 updated = strftime('%s', 'now')
-							 WHERE expires IS NULL OR expires - CAST(strftime('%s', created) AS INT) - (CAST(strftime('%s', created) AS INT) % 86400) > (? + 1) * 86400
-							 AND service_account_id IS NOT NULL;`
-		result, err := sess.Exec(rawSQL, expiryDaysLimit, expiryDaysLimit)
-		if err != nil {
-			s.log.Error("Could not update api keys", "err", err)
-		}
-		if rowsAffected, _ := result.RowsAffected(); rowsAffected > 0 {
-			s.log.Info(fmt.Sprintf("%d API keys where affected because of a new configuration value at service_accounts.token_expiration_day_limit", rowsAffected))
+
+	saTokens, err := s.ListTokens(ctx, &serviceaccounts.GetSATokensQuery{})
+	if err != nil {
+		s.log.Error("The following error has been encountered while reading the service access tokens from database.", "err", err)
+	}
+
+	threshold := time.Duration(expiryDaysLimit+1) * time.Hour * 24
+	now := time.Now()
+
+	err = s.sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
+		for _, token := range saTokens {
+			_, offset := token.Created.Zone()
+			newExpiryTime := token.Created.Truncate(time.Hour*24).Add(threshold).Unix() - int64(offset)
+			if token.Expires == nil || *token.Expires > newExpiryTime {
+				s.log.Info(fmt.Sprintf("The expiration date of token %s will be updated", strconv.Itoa(int(token.Id))))
```

Comment:
nit:
this might clog the logging a bit. Maybe consider having debug here

This will run on each startup correct? So this would always log the update for each token.

Reply:
However, keep in mind this will only be printed for tokens that have been affected. I believe that piece of information is pretty useful upon auditing why tokens "suddenly" had expiry date changes.

Reply:
yea you are right, it is only `if token.Expires == nil` so that is valid to log 👍